packages/dev/core/src/Engines/WebGPU/Extensions/engine.dynamicTexture.ts

```diff
@@ -1,11 +1,39 @@
-import type { InternalTexture } from "../../../Materials/Textures/internalTexture";
+import { ThinEngine } from "core/Engines/thinEngine";
+import { InternalTexture, InternalTextureSource } from "../../../Materials/Textures/internalTexture";
 import type { Nullable } from "../../../types";
 import { WebGPUEngine } from "../../webgpuEngine";
 import type { WebGPUHardwareTexture } from "../webgpuHardwareTexture";
 
+WebGPUEngine.prototype.createDynamicTexture = function (width: number, height: number, generateMipMaps: boolean, samplingMode: number): InternalTexture {
+    const texture = new InternalTexture(this, InternalTextureSource.Dynamic);
+    texture.baseWidth = width;
+    texture.baseHeight = height;
+
+    if (generateMipMaps) {
+        width = this.needPOTTextures ? ThinEngine.GetExponentOfTwo(width, this._caps.maxTextureSize) : width;
+        height = this.needPOTTextures ? ThinEngine.GetExponentOfTwo(height, this._caps.maxTextureSize) : height;
+    }
+
+    texture.width = width;
+    texture.height = height;
+    texture.isReady = false;
+    texture.generateMipMaps = generateMipMaps;
+    texture.samplingMode = samplingMode;
+
+    this.updateTextureSamplingMode(samplingMode, texture);
+
+    this._internalTexturesCache.push(texture);
+
+    if (width && height) {
+        this._textureHelper.createGPUTextureForInternalTexture(texture, width, height);
+    }
+
+    return texture;
+};
+
 WebGPUEngine.prototype.updateDynamicTexture = function (
     texture: Nullable<InternalTexture>,
-    canvas: HTMLCanvasElement | OffscreenCanvas,
+    source: ImageBitmap | ImageData | HTMLImageElement | HTMLCanvasElement | HTMLVideoElement | OffscreenCanvas,

```

Comment:
nit: Since this type combination is reused in a few names, maybe it should be lifted to its own type definition? Something like `WebGPUImageSources`?

Reply:
In fact, this is not a WebGPU specific type, so it should not be declared at the WebGPU level.

I could declare a type for it somewhere, but I don't really know where, as it should be independent from the graphics backend. There is `/types.ts`, but it seems to be for general types, maybe `type ImageSource = ImageBitmap | ImageData | HTMLImageElement | HTMLCanvasElement | HTMLVideoElement | OffscreenCanvas;` would be a bit too specific...

Reply:
Sounds good to me with ImageSource :-)